spynnaker/pyNN/utilities/utility_calls.py

```diff
@@ -40,7 +40,11 @@ def convert_param_to_numpy(param, no_atoms):
 
         # numpy reduces a single valued array to a single value, so enforce
         # that it is an array
-        return numpy.array([param.next(n=no_atoms)], dtype="float")
+        param_value = param.next(n=no_atoms)
+        if hasattr(param_value, '__iter__'):
+            return numpy.array(param_value, dtype="float")
+        else:
+            return numpy.array([param.next(n=no_atoms)], dtype="float")
 
```

Comment:
This doesn't look quite right; the "else" seems to call param.next again!  Guessing the issue is that pyNN 0.8 RandomDistribution.next returns an array every time, whereas pyNN 0.7 must return a single value when no_atoms = 1.  It might then be better to remove the initial "if no_atoms > 1" statement, otherwise this is becoming more cluttered for no reason.